sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/pkversion/IncrementalChangeFeedProcessorImpl.java

```diff
@@ -213,12 +214,12 @@ public Mono<Map<String, Integer>> getEstimatedLag() {
                     .flatMap(lease -> {
                         final CosmosChangeFeedRequestOptions options =
                             ModelBridgeInternal.createChangeFeedRequestOptionsForChangeFeedState(
-                                lease.getContinuationState(this.collectionId, ChangeFeedMode.INCREMENTAL));
+                                lease.getContinuationState(this.collectionResourceId, ChangeFeedMode.INCREMENTAL));
```

Comment:
This needs to be set to a `collectionResourceId` in order to generate a `collectionLink` internally.

Reply:
I don't get this comment - let's jump on a call and go through it. Seems to be a possibly breaking change after collection delete and recreate - could be that is intentional - but would like to be careful here

Reply:
for pkVersion, why we was using collectionId is because in the lease doc, we use collectionid instead collectionRid as in epkVersion

not sure how it is being used in the continuationState though, worth validating

Reply:
As discussed, will iterate on this.